WordPress/Classes/Utility/Spotlight/SearchManager.swift

```diff
@@ -431,30 +431,31 @@ fileprivate extension SearchManager {
         openListView(for: post)
         let editor = EditPostViewController.init(post: post)
         editor.modalPresentationStyle = .fullScreen
-        WPTabBarController.sharedInstance().present(editor, animated: true)
+        RootViewControllerCoordinator.sharedPresenter.rootViewController.present(editor, animated: true)
     }
 
     func openEditor(for page: Page) {
         closePreviewIfNeeded(for: page)
         openListView(for: page)
 
         let editorViewController = EditPageViewController(page: page)
-        WPTabBarController.sharedInstance().present(editorViewController, animated: false)
+        RootViewControllerCoordinator.sharedPresenter.rootViewController.present(editorViewController, animated: false)
```

Comment:
I'm thinking if we should really expose `rootViewController` from the `RootViewPresenter`.

Maybe we should hide it under the presenter interface `RootViewControllerCoordinator.sharedPresenter.present(...)`?

I understand it may be a bit problematic since I see that there're places where we access `rootViewController` directly, for example:

```swift
@objc func showAppUI(for blog: Blog? = nil, completion: Completion? = nil) {
    isShowingFullscreenSignIn = false
    show(RootViewCoordinator.sharedPresenter.rootViewController, completion: completion)
<...>
```

but since we're using a presenter here, maybe such places should be looking like this:

```swift
@objc func showAppUI(for blog: Blog? = nil, completion: Completion? = nil) {
    isShowingFullscreenSignIn = false
    RootViewCoordinator.sharedPresenter.showAppUI(on: self)
<...>
```

What are your thoughts? Did you encounter any problems by exposing `rootViewController`?

Reply:
I agree that this would be better. However, it will be tricky to make it work in some places like `ReplyTextView.btnEnterFullscreenPressed`, for example.

I'll have a go at this later on and include it in another PR if I reach something 👍

Reply:
Okay, I see. There're many cases that complicate the removal of `rootViewController` from the interface.

I agree, let's leave it outside the scope of this PR 👍.